### PR TITLE
Enable periodic principal clean up for each tenant.

### LINKED ISSUE #39: Update PII when user changed account info
## User Story
As the Insights Platform owner, I would like all PII data collected by the RBAC service to be automatically updated when users update or delete their accounts on the Customer Portal so that I can satisfy privacy requirements 

## Impacts
API backend

## Acceptance Criteria

- When a user updates their information in the Red Hat user database , any PII in RBAC database  is updated 
- When a user deletes their account, any PII is removed from the RBAC database
- The update does not have to happen immediately but within a configurable about of time e.g. within 1 business day.

### PR DESCRIPTION
**Pull request by chambridge**
Adds deployment of:
 - Redis
 - Worker
 - Scheduler

Added celery component to Django server with a single scheduled task that will run 4 times a month.
This task will sweep through all the principals in each tenant and test if they still exist. If they no longer exist they will be removed from the RBAC database along with any previous groups.

Closes #39.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -21,14 +21,46 @@ services:
...
       depends_on:
         - db
 
+  rbac-worker:
+      container_name: rbac_worker
+      build:
+          context: .
+          dockerfile: Dockerfile-env
+      working_dir: /rbac
+      entrypoint: ['celery', 'worker', '-A', 'rbac.celery', '--broker=redis://localhost:6379/0', '--loglevel=INFO']
```
Comment (dccurtis) [inline docker-compose.yml]:
```suggestion
      entrypoint: ['celery', 'worker', '-A', 'rbac.celery', '--broker=redis://redis:6379/0', '--loglevel=INFO']
```

File: docker-compose.yml [hunk 2]
```diff
@@ -21,14 +21,46 @@ services:
...
+      depends_on:
+          - redis
+  
+  rbac-scheduler:
+      container_name: rbac_scheduler
+      build:
+          context: .
+          dockerfile: Dockerfile-env
+      working_dir: /rbac
+      entrypoint: ['celery', 'beat', '-A', 'rbac.celery', '--broker=redis://localhost:6379/0', '--loglevel=INFO']
```
Comment (dccurtis) [inline docker-compose.yml]:
```suggestion
      entrypoint: ['celery', 'beat', '-A', 'rbac.celery', '--broker=redis://redis:6379/0', '--loglevel=INFO']
```

File: openshift/scheduler.yaml [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+                  name: cloudwatch
+                  optional: true
+            - name: CW_LOG_GROUP
+              valueFrom:
+                secretKeyRef:
+                  key: cw-log-group
+                  name: cloudwatch
+                  optional: true
+            - name: ROLE_SEEDING_ENABLED
+              value: "False"
```
Comment (dccurtis) [inline openshift/scheduler.yaml]: In Django settings you have this defaulting to True; same in worker.yaml

File: rbac/rbac/celery.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+# the configuration object to child processes.
+# - namespace='CELERY' means all celery-related configuration keys
+#   should have a `CELERY_` prefix.
+app.config_from_object('django.conf:settings', namespace='CELERY')
+
+app.conf.beat_schedule = {
+    'principal-cleanup-every-sevenish-days': {
+        'task': 'management.tasks.principal_cleanup',
+        'schedule':
+        crontab(0, 0, day_of_month='7-28/7'),
```
Comment (dccurtis) [inline rbac/rbac/celery.py]: 7-ish day 👍

File: openshift/scheduler.yaml (hunk 3 above)
Comment (chambridge) [reply openshift/scheduler.yaml]: Intended.

**Pull request merged**